src/libaktualizr/config/config.cc

```diff
@@ -122,7 +159,11 @@ KeyManagerConfig Config::keymanagerConfig() const {
 void Config::postUpdateValues() {
   logger_set_threshold(logger);
 
-  provision.mode = provision.provision_path.empty() ? ProvisionMode::kDeviceCred : ProvisionMode::kSharedCred;
+  if (provision.mode == ProvisionMode::kDefault) {
+    provision.mode = provision.provision_path.empty() ? ProvisionMode::kDeviceCred : ProvisionMode::kSharedCred;
+  } else if (provision.mode == ProvisionMode::kSharedCredReuse) {
+    LOG_INFO << "Provisioning mode is set to reuse shared credentials. This should only be used for testing!";
```

Comment:
Would it make sense to fail early if `provision.provision_path.empty()`? Probably, doesn't matter, really

Reply:
You mean if using shared creds but the path is empty? Maybe, but it will fail during initialization anyway. Also, nothing would stop you from still importing device creds even if you claim you want to use shared creds (I think). And if you've already provisioned and you get the URL from elsewhere, technically you don't need to keep the provisioning path configured; it'll never get used again.